src/api/transaction.ts

```diff
@@ -8,10 +9,12 @@ import { endBatch, startBatch } from "../internal"
  * @returns any value that was returned by the 'action' parameter.
  */
 export function transaction<T>(action: () => T, thisArg = undefined): T {
+    const prevAllowStateReads = allowStateReadsStart(true)
```

Comment:
Do we wanna do this? I think you should be able to supress the warning only by accessing observable from `action` or derivation.
You know, similary to strict mode (`enforceAction`). 
The low level constructs like `untracked/transaction` should be agnostic to these flags.
Eg. `useLocalStore` from `mobx-react` wraps every method inside `transaction`, because it provides batching without messing with subscriptions or `enforceAction`. 
It's explained here: https://github.com/mobxjs/mobx-react-docz/issues/12#issuecomment-532342147
So if we `allowStateReads` in `transaction` we will effectively turn of the `observableRequiresReaction` checks for all local store methods.
I know it makes the name a bit misleading - it should be `observableRequiresReactionOrAction` or even more precisely `readingObservableRequiresReactionOrAction`. Perhaps we should change it to `enforceReaction` or something? What do you think?

Reply:
I've removed that